homeassistant/components/myio/__init__.py

```diff
@@ -0,0 +1,122 @@
+
+    try:
+        hass.data[DOMAIN][_server_name] = {}
+    except (KeyError):
+        hass.data[DOMAIN] = {}
+        hass.data[DOMAIN][_server_name] = {}
+
+    hass.data[DOMAIN][_server_name]["state"] = "Offline"
+
+    try:
+        _refresh_timer = config_entry.options[CONF_REFRESH_TIME]
+    except (ValueError, Exception):  # pylint: disable=broad-except
+        _refresh_timer = 4
+
+    def server_data():
+        """Return the server data dictionary database."""
+        return hass.data[DOMAIN][_server_name]
+
+    async def setup_config_entries():
+        """Set up myIO platforms with config entry."""
+
+        # Use `hass.async_add_job` to avoid a circular dependency
+        # between the platform and the component
+        for component in PLATFORMS:
+            hass.async_add_job(
+                hass.config_entries.async_forward_entry_setup(config_entry, component)
+            )
+
+    async def async_update_data():
+        """Fetch data from API endpoint."""
+        was_offline = False
+        _temp_coordinator = hass.data[DOMAIN][_server_name]["coordinator"]
+        _temp_server_state = hass.data[DOMAIN][_server_name]["state"]
+
+        del hass.data[DOMAIN][_server_name]["coordinator"]
+        del hass.data[DOMAIN][_server_name]["state"]
+
+        if _temp_server_state == "Offline":
+            hass.states.async_set(f"{_server_name}.available", False)
+            was_offline = True
+            for component in PLATFORMS:
+                await hass.config_entries.async_forward_entry_unload(
+                    config_entry, component
+                )
+
+        # Pull fresh data from server.
+        [hass.data[DOMAIN][_server_name], _temp_server_state] = await COMMS_THREAD.send(
+            server_data=server_data(),
+            server_status=_temp_server_state,
+            config_entry=config_entry,
+            _post=None,
+        )
+
+        hass.states.async_set(f"{_server_name}.state", _temp_server_state)
+        hass.data[DOMAIN][_server_name]["state"] = _temp_server_state
+
+        if _temp_server_state.startswith("Online") and was_offline:
+            hass.states.async_set(f"{_server_name}.available", True)
+            await setup_config_entries()
+            _LOGGER.debug("Online")
+        elif not was_offline and _temp_server_state == "Offline":
+            _LOGGER.debug("PlatformNotReady")
+            hass.data[DOMAIN][_server_name]["coordinator"] = _temp_coordinator
+            raise PlatformNotReady
+
+        hass.data[DOMAIN][_server_name]["coordinator"] = _temp_coordinator
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        # Name of the data. For logging purposes.pip
+        name=f"{_server_name} status",
+        update_method=async_update_data,
+        # Polling interval.
+        update_interval=timedelta(seconds=_refresh_timer),
+    )
+
+    hass.data[DOMAIN][_server_name]["coordinator"] = coordinator
```

Comment:
Why are you recreating the coordinator on each update?

Reply:
The coordinator starts the async_update_data(). 
This pulls the new states of server, and rebuild the database. This is the only point in the integration, when it pulls the myIO server state.

Reply:
Just set new data to coordinator.data and use that instead.